### DISCUSSION TITLE
Migration from onprem ipam solutions to azure ipam

### DISCUSSION THREAD

**Discussion by iamsiri007** [Azure/ipam#284] [Ideas]
Hi there!

Is there an migrate option available to move from any other ipam solution to azure ipam? 

Currently I use phpipam hosted onprem and we are planning to move to azure ipam. We are looking for the best possible option to do so - data migration services, or any other feasible option.

**Comment by DCMattyG** [comment]
Hi there @iamsiri007, and of course thank you for reaching out!

At this point in time, there aren't any mechanisms or scripts to migrate from another IPAM platform to Azure IPAM. There are many, many IPAM platforms out there each with different features and data formats and I'm afraid it would be quite difficult to support the mapping of each of those into Azure IPAM.

That said, we're happy to help (within reason) guiding you on possible strategies on the best way to approach the situation and answering any questions about the format of data within the Azure IPAM database (Cosmos DB).

I do wish there were perhaps there were a more straight-forward way but we'll do the best we can to support you in your journey so don't hesitate to reach out with any questions you may have.

**Reply by iamsiri007** [reply]
thank you @DCMattyG  for the quick response.

We are looking to move forward by taking a dump of the mysql which is used by phpipam and then sanitize the data so that we can directly upload into the cosmo DB.

Can you share the format of the data that is needed for Azure IPAM database?

**Comment by Karl-WE** [comment]
Adding to this idea do you think it's feasible to invest into Migration from Microsoft Windows Server based IPAM feature?

The dataset is stored in SQL.
